src/rgw/rgw_common.cc

```diff
@@ -2000,15 +2000,15 @@ void RGWBucketInfo::encode(bufferlist& bl) const {
   encode(placement_rule, bl);
   encode(has_instance_obj, bl);
   encode(quota, bl);
-  encode(num_shards, bl);
-  encode(bucket_index_shard_hash_type, bl);
+  encode(layout.current_index.layout.normal.num_shards, bl);
+  encode(layout.current_index.layout.normal.hash_type, bl);
```

Comment:
we're already encoding these in `encode(layout, bl);` below, so we should remove these fields from the encoding here. that will complicate the decode below though. for example:
```diff
+  static constexpr uint8_t new_layout_v = 22;
-  if (struct_v >= 10)
+  if (struct_v >= 10 && struct_v < new_layout_v)
     decode(layout.current_index.layout.normal.num_shards, bl);
-  if (struct_v >= 11)
+  if (struct_v >= 11 && struct_v < new_layout_v)
     decode(layout.current_index.layout.normal.hash_type, bl);
...
-  if (struct_v >= 15) {
+  if (struct_v >= 15 && struct_v < new_layout_v)
```